extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/LoadedSegmentDataProvider.java

```diff
@@ -131,8 +149,12 @@ public <RowType, QueryType> Pair<DataServerQueryStatus, Yielder<RowType>> fetchR
 
     Pair<DataServerQueryStatus, Yielder<RowType>> statusSequencePair;
     try {
+      // We need to check for handoff to decide if we need to retry. Therefore, we handle it here instead of inside
+      // the client.
       statusSequencePair = RetryUtils.retry(
           () -> {
+            ServiceLocation serviceLocation = Iterables.getOnlyElement(fixedSetServiceLocator.locate().get().getLocations());
```

Comment:
Nit: Use collectionUtils.getOnlyElement() so that a nicer error message is thrown.

Reply:
Changed